File: 02-backend/spring-boot-eccomerce/src/main/java/com/annacode/eccomerce/dto/Purchase.java
package com.annacode.eccomerce.dto;

import com.annacode.eccomerce.entity.Address;
import com.annacode.eccomerce.entity.Customer;
import com.annacode.eccomerce.entity.Order;
import com.annacode.eccomerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase  {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;

}

File: 03-frontend/angular-ecommerce/src/app/common/purchase.ts
import { Customer } from './customer';
import { OrderItem } from './order-item';
import { Order } from './order';
import { Address } from './customerAddress';

export class Purchase {
  customer!: Customer;
  shippingAddress!: Address;
  billingAddress!: Address;
  order!: Order;
  orderItems!: OrderItem[];
}
